pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -121,17 +121,23 @@ func (lb *ListenerBuilder) buildVirtualOutboundListener() *ListenerBuilder {
 
 	filterChains := buildOutboundCatchAllNetworkFilterChains(lb.node, lb.push)
 
-	actualWildcard, _ := getActualWildcardAndLocalHost(lb.node)
+	oWildcardAndLocalHost := NewHostAddresses(lb.node.GetIPMode())
```

Comment:
I think instead of constructing each time, we can define several vars mapping to IP mode

Reply:
This was one of my suggestions of adding it to node structure. So that majority of cases we are covered, we compute only when we detect IP mode

Reply:
Hi @hzxuzhonghu & @ramaraochavali,  besides the service with proxy, we also need to think about the service without proxy, so we may need to balance. Any idea?

Reply:
Not sure what you mean, a service without proxy can not be run with this function

Reply:
I get that part. But I think having it node will help majority cases as I explained.  For others we can build on the fly. Also buildVirtualOutboundListener is for regular proxies.

Reply:
Hi @ramaraochavali & @hzxuzhonghu,  as you required, I move the `HostAddresses` logic into proxy node now. And it's okay to build the abnormal situation on the fly. Thanks for your comments! ^_^